adoc/admin-security-certificates.adoc

```diff
@@ -0,0 +1,208 @@
+= Certificates
+
+During the installation of {productname}, a CA (Certificate Authority) certificate is generated; that is then used to authenticate and verify all communications. The process also creates and distributes client certificates for the components.
+
+Communication is secured with TLS v1.2 using the AES 128 CBC cipher. All client certificates are 4096 Bit RSA encrypted.
+
+The CA certificate is valid for 3650 days (10 years) by default. The client certificates are valid for 365 days (1 year) by default.
+
+Certificates are located in `/etc/kubernetes/pki` on each master nodes. Three root CA required for {productname} are
+
+* {kube} apiserver: path `/etc/kubernetes/pki/ca.{crt.key}`
+* front-end proxy: path `/etc/kubernetes/pki/front-proxy-ca.{crt.key}`
+* etcd: path `/etc/kubernetes/pki/etcd/ca.{crt.key}`
+
+The following certificates are managed by `kubeadm`.
+
+[%header,cols=4*]
+|===
+|Common Name
+|Parent CA
+|Path
+|Kind
+
+|kubernetes
+|
+|ca.crt,key
+|CA
+
+|kube-apiserver
+|kubernetes
+|apiserver.crt,key
+|Server
+
+|kube-apiserver-etcd-client
+|kubernetes
+|apiserver-etcd-client.crt,key
+|Client
+
+|kube-apiserver-kubelet-client
+|kubernetes
+|apiserver-kubelet-client.crt,key
+|Client
+
+|front-proxy-ca
+|
+|front-proxy-ca.crt,key
+|CA
+
+|front-proxy-client
+|front-proxy-ca
+|front-proxy-client.crt,key
+|Client
+
+|etcd-ca
+|
+|etcd/ca.crt,key
+|CA
+
+|kube-etcd-healthcheck-client
+|etcd-ca
+|etcd/healthcheck-client.crt,key
+|Client
+
+|kube-etcd-peer
+|etcd-ca
+|etcd/peer.crt,key
+|Server,Client
+
+|kube-etcd-server
+|etcd-ca
+|etcd/server.crt,key
+|Server,Client
+|===
+
+The following certificates are created by `skuba` and are stored in `Secret` resource.
+
+[%header,cols=4*]
+|===
+|Common Name
+|Parent CA
+|Secret Resource Name
+|Kind
+
+|oidc-dex
+|kubernetes
+|oidc-dex-cert
+|Server
+
+|oidc-gangway
+|kubernetes
+|oidc-gangway-cert
+|Server
+
+|cilium-etcd-client
+|etcd-ca
+|cilium-secret
+|Client
+|===
+
+== Deployment with custom root CA certificate
+
+Administrators can provide custom root CA certificates during cluster deployment and administrators can decide which root CA components to replace or replace all.
```

Comment:
Suggest:

Administrators can provide custom root CA certificates during cluster deployment and decide which root CA components to replace or replace all.